portfolio/src/main/webapp/typing.js

```diff
@@ -0,0 +1,58 @@
+// Type Writer JS Animation
+
+class TyperWriter {
+  constructor(elements, words, waitTime = 3000) {
+    this.elements = elements;
+    this.words = words;
+    this.wordIndex = 0;
+    this.txt = "";
+    this.waitTime = parseInt(waitTime, 10); //Base 10
+    this.typeFunc();
+    this.isDelete = false;
+  }
+}
+//typing main function
+TyperWriter.prototype.typeFunc = function () {
+  console.log(this.words.length);
+  const current_index = this.wordIndex % this.words.length;
+  const full_word = this.words[current_index];
+
+  if (this.isDelete) {
+    //Remove character
+    this.txt = full_word.substring(0, this.txt.length - 1);
+  } else {
+    //Add character
+    this.txt = full_word.substring(0, this.txt.length + 1);
+  }
+
+  //Output on website
+  this.elements.innerHTML = '<span class = "text">' + this.txt + "</span>";
+  let typeSpeed = 100;
+  if (this.isDelete) {
+    typeSpeed /= 2;
+  }
+  //Change type speed when deleting
+  if (!this.isDelete && this.txt == full_word) {
+    typeSpeed = this.waitTime;
+    this.isDelete = true;
+  } else if (this.isDelete && this.txt == "") {
+    this.isDelete = false;
+    this.wordIndex += 1;
+    typeSpeed = 500;
+  }
+
+  setTimeout(() => this.typeFunc(), typeSpeed);
+};
+
+//initialize object
```

Comment:
This comment is unhelpful; it's better to remove.

Comments should describe "why" or any caveats, never "what." Also, if you comment "what" code does, when a programmer changes the code, they often forget to change the comment leaving it out of date and misleading. See section 3.6.1 of this book:

https://books.google.com/books?id=fLcPBwAAQBAJ&pg=PA45&lpg=PA45&dq=increment+1+commenting+style&source=bl&ots=D6KmUjCjFD&sig=ACfU3U2MQL4715oBi1wxewS027DLisPzdA&hl=en&sa=X&ved=2ahUKEwij1tucwJ3qAhWGJzQIHSM4DE8Q6AEwAHoECAcQAQ#v=onepage&q&f=false